django/thunderstore/repository/models/namespace.py

```diff
@@ -0,0 +1,42 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.db.models import Manager
+
+from thunderstore.repository.validators import PackageReferenceComponentValidator
+
+
+class Namespace(models.Model):
+    objects: "Manager[Namespace]"
+
+    name = models.CharField(
```

Comment:
This can be made the primary key

Reply:
I'd rather not cause then the validate functions, "name is read only" will be hard to implement. (Or at least I didn't find any other good way to check if the object is in the database, unless we pass the old name to the save function always)

Reply:
That's exactly why

https://docs.djangoproject.com/en/dev/ref/models/fields/#primary-key
> The primary key field is read-only. If you change the value of the primary key on an existing object and then save it, a new object will be created alongside the old one.